speed-dreams/source-2.2.3/src/simulatedDrivingAssistance/backendUnitTests/RecorderTests.cpp

```diff
@@ -439,102 +440,90 @@ void TestV0ToV1Changes(void* p_upgradedRunSettingsHandle, filesystem::path& p_to
+    {
+        case 1:
+            AssertV0ToV1Changes(p_upgradedRunSettingsHandle, p_toUpgrade);
+            break;
+        case 2:
+            AssertV1ToV2Changes(p_upgradedRunSettingsHandle);
+            break;
+        case 3:
+            AssertV2ToV3Changes(p_upgradedRunSettingsHandle);
+            break;
+        case 4:
+            AssertV3ToV4Changes(p_upgradedRunSettingsHandle);
+            break;
+        default:
+            SUCCEED();
+    }
 }
 
+/// @brief TestFixture to help with testing the recording upgrades in a systematic way.
+class RecorderUpgradeVersionTestFixture : public ::testing::TestWithParam<int>
+{
+};
 
-/// @brief Test whether the upgrade from V2 is done correctly
-TEST(RecorderTests, UpgradeFromV2Test)
+/// @brief Tests whether the base-recording is succesfully upgrade to the target version.
+TEST_P(RecorderUpgradeVersionTestFixture, UpgradeToVersion)
 {
-    INIT_VALIDATE_OR_UPGRADE_TEST("v2-recording", toUpgrade)
+    int targetVersion = GetParam();
 
-    ASSERT_TRUE(Recorder::ValidateAndUpdateRecording(toUpgrade));
+    // Start upgrading from the base v0 recording.
+    INIT_VALIDATE_OR_UPGRADE_TEST("v0-recording", toUpgrade);
+    ASSERT_TRUE(Recorder::ValidateAndUpdateRecording(toUpgrade, targetVersion));
 
+    // Check whether the updated recording settings were correctly created.
     void* upgradedRunSettingsHandle = GfParmReadFile(filesystem::path(toUpgrade).append(RUN_SETTINGS_FILE_NAME).string().c_str(), 0, true);
-
     ASSERT_NE(upgradedRunSettingsHandle, nullptr);
 
-    // Now on latest version
-    ASSERT_EQ(GfParmGetNum(upgradedRunSettingsHandle, PATH_VERSION, KEY_VERSION, nullptr, NAN), CURRENT_RECORDER_VERSION);
+    // Now on the targeted version
+    ASSERT_EQ(GfParmGetNum(upgradedRunSettingsHandle, PATH_VERSION, KEY_VERSION, nullptr, NAN), targetVersion);
 
-    // Test that all changes that need to be made from this version are present
-    TestV2ToV3Changes(upgradedRunSettingsHandle);
+    // Check whether the changes made in the target version are contained.
+    AssertTargetVersionChanges(upgradedRunSettingsHandle, toUpgrade, targetVersion);
 }
 
-/// @brief Test whether the latest version of recording contains all correct elements
-/// @note Currently on version 3
-TEST(RecorderTests, ValidateLatestRecordingTest)
-{
-    INIT_VALIDATE_OR_UPGRADE_TEST("v3-recording", toUpgrade);
-
-    ASSERT_TRUE(Recorder::ValidateAndUpdateRecording(toUpgrade));
-
-    ASSERT_TRUE(filesystem::exists(filesystem::path(toUpgrade).append(USER_INPUT_RECORDING_FILE_NAME)));
-    ASSERT_TRUE(filesystem::exists(filesystem::path(toUpgrade).append(SIMULATION_DATA_RECORDING_FILE_NAME)));
-    ASSERT_TRUE(filesystem::exists(filesystem::path(toUpgrade).append(DECISIONS_RECORDING_FILE_NAME)));
+/// @brief       Generates all the tests for the recording upgrades
+/// @param Range The versions to upgrade the recordings to, is a range from [1..RECORDER_VERSION]
+INSTANTIATE_TEST_CASE_P(
```

Comment:
INSTANTIATE_TEST_CASE_P seems to be deprecated. It says to use INSTANTIATE_TEST_SUITE_P